sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -446,6 +688,9 @@ static const struct hda_dai_widget_dma_ops hda_ipc4_dma_ops = {
 	.codec_dai_set_stream = hda_codec_dai_set_stream,
 	.calc_stream_format = hda_calc_stream_format,
 	.get_hlink = hda_get_hlink,
+	.set_up_be_pipeline = hda_ipc4_set_up_be_pipeline,
+	.free_be_pipeline = hda_ipc4_free_be_pipeline,
+	.unprepare_be_pipeline = hda_dai_unprepare_widgets_in_pipeline,
```

Comment:
So, if the BE is not serviced by HD-DMA (but with dwdma or something else) and not IPC4 I assume then we don't need to do this black magic?